src/ostream

```diff
@@ -27,6 +27,11 @@
 #include "cstdio"
 #include "ostream_helpers"
 
+#ifdef ARDUINO_ARCH_AVR
```

Comment:
I would consider leaving out these include guards, since pretty much every (non-AVR) Arduino core provides dummy version of the `F()` macro and `pgm_read_byte` and friends, so adding this code unconditionally makes user code that uses `F()` and ArduinoSTL more portable.

Reply:
Ok, but this code will not run on other than AVR. avr/pgmspace.h is required to read from flash.

Reply:
> Ok, but this code will not run on other than AVR. avr/pgmspace.h is required to read from flash.

Most cores actually provide a dummy `avr/pgmspace.h`, so portable sketches can still use `F()` (etc.) and compile on non-AVR too. E.g. see https://github.com/arduino/ArduinoCore-sam/tree/master/cores/arduino/avr and https://github.com/stm32duino/Arduino_Core_STM32/tree/main/cores/arduino/avr (better to just include `Arduino.h` instead, though, since that should also include pgmspace or its dummy).

Reply:
I understand. Code for SAM and STM should compile without complaning about missing F() macro